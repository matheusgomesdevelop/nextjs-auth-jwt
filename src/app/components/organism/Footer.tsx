import React from 'react';

import Icon from '../atoms/Icon';
import Link from '../atoms/Link';

const Footer: React.FC = () => {
    const date = new Date();

    return (
        <footer className="container flex justify-between items-center w-full">
            <p className="text-gray-500 flex gap-6">
                Â© {date.getFullYear()}
                <Link href={'https://matheusgomesdev.com.br'}>matheusgomesdev.</Link>
            </p>

            <div className="flex gap-24">
                <Link href={'https://github.com/matheusgrodrigues/nextjs-auth-jwt'}>
                    <Icon icon="pi-github" />
                </Link>

                <Link href={'https://www.linkedin.com/in/matheusgomes/'}>
                    <Icon icon="pi-linkedin" />
                </Link>
            </div>
        </footer>
    );
};

export default Footer;
