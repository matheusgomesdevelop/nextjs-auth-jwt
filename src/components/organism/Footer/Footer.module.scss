@import "../../../styles/util/_breakpoints.scss";

.footer {
  width: 100%;
  min-height: 88px;
  max-width: 1280px;
  padding: 0 var(--spacing-32);
  @include centralizar;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  align-items: center;
  gap: var(--spacing-32);

  @include d(tablet) {
    flex-direction: row;
    height: 88px;
    gap: 0;
  }

  &__name {
    & > p {
      display: flex;
      gap: var(--spacing-6);
      color: var(--gray-500);
    }
  }

  &__icons {
    display: flex;
    gap: var(--spacing-24);

    & i {
      color: var(--gray-500);
    }
  }
}
