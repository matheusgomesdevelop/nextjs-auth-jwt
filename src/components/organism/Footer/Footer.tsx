import React from 'react';

import styles from './Footer.module.scss';

import { Link, Icon } from '@/components/atoms';

interface FooterProps {
    name: string;
    site: string;
    github: string;
    linkedin: string;
}

const Footer: React.FC<FooterProps> = ({ name, site, github, linkedin }) => {
    const date = new Date();

    return (
        <div className={styles.footer} id="o-footer">
            <div className={styles.footer__name}>
                <p>
                    Â© {date.getFullYear()}
                    <Link href={site}>{name}.</Link>
                </p>
            </div>

            <div className={styles.footer__icons}>
                <Link href={github}>
                    <Icon icon="pi-github" />
                </Link>

                <Link href={linkedin}>
                    <Icon icon="pi-linkedin" />
                </Link>
            </div>
        </div>
    );
};

export default Footer;
