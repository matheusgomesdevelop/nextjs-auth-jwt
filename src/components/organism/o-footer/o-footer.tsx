// StyleSheet
import styles from "./o-footer.module.scss";

// Atoms
import { A_Icon } from "@/components/atoms/a-icon/a-icon";
import { A_Link } from "@/components/atoms/a-link/a-link";

export interface I_OFooter {
  name: string;
  site: string;
  github: string;
  linkedin: string;
}

export const O_Footer = ({ name, site, github, linkedin }: I_OFooter) => {
  const date = new Date();

  return (
    <footer className={styles.o_footer}>
      <div className={styles.o_footer__name}>
        <p>
          Â© {date.getFullYear()}
          <A_Link href={site}>{name}</A_Link>.
        </p>
      </div>

      <div className={styles.o_footer__icons}>
        <A_Link href={github}>
          <A_Icon icon="pi-github" />
        </A_Link>

        <A_Link href={linkedin}>
          <A_Icon icon="pi-linkedin" />
        </A_Link>
      </div>
    </footer>
  );
};
