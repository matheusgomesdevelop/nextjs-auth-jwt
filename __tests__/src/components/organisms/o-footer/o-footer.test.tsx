import renderer from 'react-test-renderer';

// Organism
import { O_Footer } from '@/components/organism/Footer/o-footer';

describe('Deve renderizar o footer corretamente', () => {
    it('Deve preservar a estrutura visual do footer', () => {
        const three = renderer
            .create(
                <O_Footer
                    name="matheusgomesdev"
                    site="https://matheusgomesdev.com.br"
                    github="https://github.com/matheusgrodrigues"
                    linkedin="https://www.linkedin.com/in/matheusgomes/"
                />
            )
            .toJSON();

        expect(three).toMatchSnapshot();
    });
});
