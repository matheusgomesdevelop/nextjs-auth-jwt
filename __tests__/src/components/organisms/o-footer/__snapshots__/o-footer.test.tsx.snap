// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Deve renderizar o footer corretamente Deve preservar a estrutura visual do footer 1`] = `
<footer
  className="o_footer"
  id="o-footer"
>
  <div
    className="o_footer__name"
  >
    <p>
      Â© 
      2023
      <a
        className="a_link"
        href="https://matheusgomesdev.com.br"
      >
        matheusgomesdev
        .
      </a>
    </p>
  </div>
  <div
    className="o_footer__icons"
  >
    <a
      className="a_link"
      href="https://github.com/matheusgrodrigues"
    >
      <i
        className="a_icon pi pi-github"
      />
    </a>
    <a
      className="a_link"
      href="https://www.linkedin.com/in/matheusgomes/"
    >
      <i
        className="a_icon pi pi-linkedin"
      />
    </a>
  </div>
</footer>
`;
